src/components/task/TaskPanel.tsx

```diff
@@ -0,0 +1,186 @@
+    border: '1px solid rgba(0, 0, 0, .125)',
+    boxShadow: "none"
+  }
+})(MuiExpansionPanel);
+
+interface State {
+  hasFailingTests: boolean
+  hasOtherTests: boolean
+  failingTests: APITest[]
+  otherTests: APITest[]
+  isShowingOtherTests: boolean
+}
+
+class Props {
+  public client: rest.Evergreen
+  public task: APITask
+  public status: string
+  public switchTask: (task: APITask) => void
+  public isCurrentTask: boolean
+  public onFinishStateUpdate: () => void;
+}
+
+export class TaskPanel extends React.Component<Props, State> {
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      hasFailingTests: false,
+      hasOtherTests: false,
+      failingTests: [],
+      otherTests: [],
+      isShowingOtherTests: false
+    }
+  }
+
+  public componentDidMount() {
+    this.props.client.getTestsForTask((err, resp, body) => {
+      const tests = ConvertToAPITests(body) as unknown as APITest[];
+      let failingTests: APITest[] = [];
+      let otherTests: APITest[] = [];
+      Array.from(tests).forEach(element => {
+        if (element.status.includes("fail")) {
+          failingTests.push(element);
+        } else {
+          otherTests.push(element);
+        }
+      });
+      failingTests = failingTests.sort(this.sortByTestFile);
+      otherTests = otherTests.sort(this.sortByTestFile);
+      this.setState({
+        hasFailingTests: failingTests.length !== 0 ? true : false,
+        hasOtherTests: otherTests.length !== 0 ? true : false,
+        failingTests: failingTests,
+        otherTests: otherTests
+      })
+    }, this.props.task.task_id);
+  }
+
+  public render() {
+
+    const FailedTests = () => (
+      <Grid container={true} direction="column" spacing={1}>
+        {this.state.hasFailingTests === true ?
+          this.state.failingTests.map(test => (
+            <Grid item={true} xs={12} key={test.test_file}>
+              <Typography color="error">
+                {test.test_file.split("/").pop()}
+              </Typography>
+            </Grid>
+          )) :
+          <Grid item={true} xs={12}>
+            <Typography className="no-tests-text">
+              No failing tests
+            </Typography>
+          </Grid>
+        }
+      </Grid>
+    );
+
+    const OtherTests = () => (
+      <Card>
+        <CardActions>
+        <IconButton onClick={this.handleOtherTestsClick}>
+          {this.state.isShowingOtherTests ? <ExpandLessIcon/> : <ExpandMoreIcon />}
+          </IconButton>
+          <Typography>
+            Other Tests
+          </Typography>
+        </CardActions>
+        <Collapse in={this.state.isShowingOtherTests} timeout="auto" unmountOnExit={true}>
+          <CardContent className="other-tests-list">
+            <Grid container={true} spacing={1}>
+              {this.state.otherTests.map(test => (
+                <Grid item={true} xs={12} key={test.test_file}>
+                  <Typography>
+                    {test.test_file.split("/").pop()}
```

Comment:
I sort of expected clicking a test would show me the test logs in the panel. Are you planning on implementing that later?

Reply:
Yes, I was. There's kind of a lingering design question there- because the test logs are linked through LogKeeper, I'm not totally sure how to retrieve or display them.

Reply:
You can retrieve them the same way. The link property of the task will just point to logkeeper, which should be able to serve up the log in plain text format as well

Reply:
Done